smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -21,6 +20,24 @@ contract MixinPurchase is
 {
   event RenewKeyPurchase(address indexed owner, uint newExpiration);
 
+  event GasRefunded(address indexed receiver, uint refundedAmount, address tokenAddress);
+
+  uint8 private _gasRefundPercentage = 0; // default to 0
+
+  /**
+  * @dev Set a percentage of the key price to be refunded to the sender on purchase
+  */
+  function setGasRefundPercentage(uint8 _percent) external onlyLockManagerOrBeneficiary {
```

Comment:
```suggestion
  function setGasRefundPercentage(uint8 _percent) external onlyLockManager {
```

Let's limit this to the lock manager.

Reply:
Also we should probably check that it is lower (or equal even tho that would not make a lot of sense) to 10,000.